Makefile

```diff
@@ -8,14 +8,31 @@ COMPARE ?= 1
 NON_MATCHING ?= 0
 # If ORIG_COMPILER is 1, compile with QEMU_IRIX and the original compiler
 ORIG_COMPILER ?= 0
+# If COMPILER is "gcc", compile with GCC instead of IDO.
+COMPILER ?= ido
+
+CFLAGS ?=
+CPPFLAGS ?=
+
+# ORIG_COMPILER cannot be combined with a non-IDO compiler. Check for this case and error out if found.
+ifneq ($(COMPILER),ido)
+  ifeq ($(ORIG_COMPILER),1)
+    $(error ORIG_COMPILER can only be used with the IDO compiler. Please check your Makefile variables and try again)
+  endif
+endif
+
+ifeq ($(COMPILER),gcc)
+  CPPFLAGS += -DCOMPILER_GCC
```

Comment:
Just a minor suggestion but `CFLAGS` could probably be set to define COMPILER_GCC here as well, for consistency and to avoid confusion in case someone tries to use it in the codebase